### ISSUE TITLE
Regression for `valid-title` and `prefer-hooks-on-top` when using `test.scoped`

### ISSUE THREAD

**Issue by acidoxee** [vitest-dev/eslint-plugin-vitest#833]
Hi, first of all, thanks for this fantastic plugin!

Since v1.4.0, there seems to be a regression for some rules when using `test.scoped`. From what I can tell in my own codebase, at least `valid-title` and `prefer-hooks-on-top` are impacted.

This is a short example that resembles some of my test files:

```ts
test.scoped({ seed: true });

beforeEach(async ({ db, msw }) => {
	// ...
});

it('should do something', async ({ db }) => {
	// ...
});
```

I've got no issues on this kind of file when using v1.3.26 of the plugin, but when bumping to v1.4.0 or higher (I've confirmed it's still an issue on v1.5.0), there are the following issues:

- `test.scoped` is flagged with the error `Test title must be a string, a function or class name (vitest/valid-title)`
- The `beforeEach` hook is flagged with the error `Hooks should come before test cases (vitest/prefer-hooks-on-top)`

Do note that if I move `test.scoped` after `beforeEach`, but still before the test cases, the issue for `prefer-hooks-on-top` disappears, but the one for `valid-title` is still present. I'm guessing this special usage of `test.scoped` is currently flagged as a test, hence rules regarding tests and hook order being confused.
